### PR TITLE
[fix](auth)fix create table like need create_priv of existed table

### PR DESCRIPTION
**Pull request by zddr**
create table like only need create_priv of newTable, and select_priv of existedTable

not need create_priv of existed table

ps: have already added cases in other PRs

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run cloud_p0

**Comment by zddr** [comment]
run cloud_p0

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java [hunk 1]
```diff
@@ -1259,14 +1259,22 @@ public void createTableLike(CreateTableLikeStmt stmt) throws DdlException {
...
-            CreateTableStmt parsedCreateTableStmt = (CreateTableStmt) SqlParserUtils.parseAndAnalyzeStmt(
-                    createTableStmt.get(0), ConnectContext.get());
-            parsedCreateTableStmt.setTableName(stmt.getTableName());
-            parsedCreateTableStmt.setIfNotExists(stmt.isIfNotExists());
-            createTable(parsedCreateTableStmt);
+            try {
+                // analyze CreateTableStmt will check create_priv of existedTable, create table like only need
+                // create_priv of newTable, and select_priv of existedTable, and priv check has done in
+                // CreateTableStmt/CreateTableCommand, so we skip it
+                ConnectContext.get().setSkipAuth(true);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:1266]: check ConnectContext.get() != null

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java [hunk 2]
```diff
@@ -1259,14 +1259,27 @@ public void createTableLike(CreateTableLikeStmt stmt) throws DdlException {
...
+            ConnectContext ctx = ConnectContext.get();
+            try {
+                // analyze CreateTableStmt will check create_priv of existedTable, create table like only need
+                // create_priv of newTable, and select_priv of existedTable, and priv check has done in
+                // CreateTableStmt/CreateTableCommand, so we skip it
+                if (ctx != null) {
+                    ctx.setSkipAuth(true);
+                }
+                CreateTableStmt parsedCreateTableStmt = (CreateTableStmt) SqlParserUtils.parseAndAnalyzeStmt(
+                        createTableStmt.get(0), ConnectContext.get());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:1271]: could parseAndAnalyzeStmt  work well when `ConnectContext.get()` is null ?
```suggestion
                CreateTableStmt parsedCreateTableStmt = (CreateTableStmt) SqlParserUtils.parseAndAnalyzeStmt(
                        createTableStmt.get(0), ctx);
```

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run p0

**Review by morrySnow — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**